foxtrot-core/src/main/java/com/flipkart/foxtrot/core/querystore/actions/GroupAction.java

```diff
@@ -83,7 +83,11 @@
 
     public GroupAction(GroupRequest parameter, String cacheToken, AnalyticsLoader analyticsLoader) {
         super(parameter, cacheToken, analyticsLoader);
-        emailClient = getEmailClient(analyticsLoader.getEmailConfig());
+        if (analyticsLoader.getEmailClient() != null){
+            emailClient = analyticsLoader.getEmailClient();
+        }else {
```

Comment:
Formatting issue. This is the worst possible place to do this. Email can be sent for a variety of reasons. Please have either use:
- a Guava MessageBus to send out events on which send out mails from a subscriber
- create a listener interface and implement that to send email.

Reply:
Will make these changes